File: proselint/checks/industrial_language/jargon.py
"""
Jargon.

---
layout:     post
source:     Garner's Modern American Usage
source_url: http://bit.ly/1T4alrY
title:      Jargon
date:       2014-06-10
categories: writing
---



"""

from proselint.registry.checks import Check, types

check = Check(
    check_type=types.Existence(
        items=(
            "in the affirmative",
            "in the negative",
            "agendize",
            "per your order",
            "per your request",
            "disincentivize",
        )
    ),
    path="industrial_language.jargon",
    message="'{}' is jargon. Can you replace it with something more standard?",
)

__register__ = (check,)

File: proselint-rs/crates/proselint-checks/src/industrial_language/jargon.rs
use proselint_registry::checks::{Check, types::*, Padding};

pub const EXAMPLES_PASS: &[&str] = &["Smoke phrase with nothing flagged."];
pub const EXAMPLES_FAIL: &[&str] = &["I agree it's in the affirmative."];

const CHECK: Check = Check {
	check_type: &Existence {
		items: &[
			"in the affirmative",
			"in the negative",
			"agendize",
			"per your order",
			"per your request",
			"disincentivize",
		],
		padding: Padding::WordsInText,
		exceptions: &[],
	},
	path: "industrial_language.jargon",
	msg: "'{}' is jargon. Can you replace it with something more standard?",
	..Check::default()
};

pub const REGISTER: &[Check] = &[CHECK];
